### ISSUE TITLE
Server Fails to Start 50% of the Time After Upgrading to Version 2.6.0

### ISSUE THREAD

**Issue by CrazyNutLz** [GTNewHorizons/GT-New-Horizons-Modpack#16141]
### Your GTNH Discord Username

crazynutlz

### Your Pack Version

2.6.0

### Your Server

private server

### Java Version

Java 21

### Type of Server

Crucible

### Your Actions

Server Fails to Start 50% of the Time After Upgrading to Version 2.6.0

When the server starts up, it often crashes around the point where the log shows the following message: "[GregTech GTNH]: BulkRemoveByRecipe: tList: 22336 toRemove: 10936." There's also a chance that the server crash will cause my computer to reboot. Attached are the logs from both successful and unsuccessful startups. Could someone help me identify what might be causing this issue? Thank you.

### Crash Report

Successful Startup Log: 
[suc.txt](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/files/15199224/suc.txt)

Error Log: 
[error.txt](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/files/15199225/error.txt)

[hs_err_pid15120.log](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/files/15199289/hs_err_pid15120.log)

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying the crash still exists will prompt us to investigate and reopen it once we confirm your report.
- [ ] I can reproduce this crash consistently by following the exact steps I described above.
- [ ] I have asked other people and they confirm they also crash by following the exact steps I described above.

**Comment by eigenraven** [comment]
The error log ends with
```
# An error report file with more information is saved as:
# G:\0---------�����---------0\GT_New_Horizons_2.3.0_Server\hs_err_pid15120.log
```
can you post that extra log file too?

**Comment by boubou19** [comment]
try booting with a plain forge server and see if it increases the boot rate. We don't support crucible

**Comment by eigenraven** [comment]
` -javaagent:nide8auth.jar=a000d3f85bc311ea908800163e095b49 ` can you explain what nide8auth is? It's registered as a java agent that can rewrite any class in the game and java itself, and sounds like a custom authentication provider that would enable game piracy (which we do not support).

**Comment by CrazyNutLz** [comment]
> `-javaagent:nide8auth.jar=a000d3f85bc311ea908800163e095b49` can you explain what nide8auth is? It's registered as a java agent that can rewrite any class in the game and java itself, and sounds like a custom authentication provider that would enable game piracy (which we do not support).

https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/16141#issuecomment-2092818028

**Comment by CrazyNutLz** [comment]
> > `-javaagent:nide8auth.jar=a000d3f85bc311ea908800163e095b49` can you explain what nide8auth is? It's registered as a java agent that can rewrite any class in the game and java itself, and sounds like a custom authentication provider that would enable game piracy (which we do not support).
> 

This is a login tool for my server, and the crash should not be related to this. The crash occurs later when it loads GregTech.

**Comment by boubou19** [comment]
try without it pls

**Comment by CrazyNutLz** [comment]
> try without it pls

Thanks for the reply, I will test it several more times later.

**Comment by CrazyNutLz** [comment]
I tried to start with the default parameters again, without making any changes, and there's still a chance it gets stuck at the same place. Sometimes it just freezes without generating any logs, as shown in the picture. Here is the log file.

Thanks a lot.
@boubou19 @eigenraven 

![QQ截图20240504003756](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/assets/43942933/31b96d7f-4e47-4d19-9529-756bb891dbb7)

[hs_err_pid8448.log](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/files/15202770/hs_err_pid8448.log)
[cmd_log.txt](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/files/15202773/cmd_log.txt)

**Comment by eigenraven** [comment]
```
J 795  jdk.internal.misc.Unsafe.allocateInstance(Ljava/lang/Class;)Ljava/lang/Object; java.base@21.0.3 (0 bytes) @ 0x000001eb627303a7 [0x000001eb62730340+0x0000000000000067]
J 23661 c1 gregtech.api.util.GT_ModHandler.lambda$bulkRemoveByRecipe$2(Ljava/util/List;Lnet/minecraft/item/crafting/IRecipe;)Z (39 bytes) @ 0x000001eb5d521864 [0x000001eb5d5212a0+0x00000000000005c4]
J 23662 c1 gregtech.api.util.GT_ModHandler$$Lambda+0x000001eb0355ac50.test(Ljava/lang/Object;)Z (12 bytes) @ 0x000001eb5d5225c4 [0x000001eb5d522440+0x0000000000000184]
J 10193 c2 java.util.stream.ReferencePipeline$2$1.accept(Ljava/lang/Object;)V java.base@21.0.3 (27 bytes) @ 0x000001eb63361880 [0x000001eb63361820+0x0000000000000060]
J 19400 c2 java.util.ArrayList$ArrayListSpliterator.forEachRemaining(Ljava/util/function/Consumer;)V java.base@21.0.3 (127 bytes) @ 0x000001eb63b3acac [0x000001eb63b3abc0+0x00000000000000ec]
J 11073 c2 java.util.stream.AbstractPipeline.wrapAndCopyInto(Ljava/util/stream/Sink;Ljava/util/Spliterator;)Ljava/util/stream/Sink; java.base@21.0.3 (18 bytes) @ 0x000001eb62ba17d4 [0x000001eb62ba1660+0x0000000000000174]
J 23484 c1 java.util.stream.ReduceOps$ReduceTask.doLeaf()Ljava/lang/Object; java.base@21.0.3 (5 bytes) @ 0x000001eb5c84a50c [0x000001eb5c84a2e0+0x000000000000022c]
J 23487 c1 java.util.stream.AbstractTask.compute()V java.base@21.0.3 (141 bytes) @ 0x000001eb5ad9af64 [0x000001eb5ad9a800+0x0000000000000764]
J 14111 c1 java.util.concurrent.CountedCompleter.exec()Z java.base@21.0.3 (6 bytes) @ 0x000001eb5bf46944 [0x000001eb5bf46840+0x0000000000000104]
J 23480 c1 java.util.concurrent.ForkJoinTask.doExec()I java.base@21.0.3 (37 bytes) @ 0x000001eb5ba281cc [0x000001eb5ba28080+0x000000000000014c]
j  java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Ljava/util/concurrent/ForkJoinTask;Ljava/util/concurrent/ForkJoinPool$WorkQueue;)V+19 java.base@21.0.3
J 14112 c1 java.util.concurrent.ForkJoinPool.scan(Ljava/util/concurrent/ForkJoinPool$WorkQueue;II)I java.base@21.0.3 (263 bytes) @ 0x000001eb5c169274 [0x000001eb5c168c40+0x0000000000000634]
j  java.util.concurrent.ForkJoinPool.runWorker(Ljava/util/concurrent/ForkJoinPool$WorkQueue;)V+35 java.base@21.0.3
j  java.util.concurrent.ForkJoinWorkerThread.run()V+31 java.base@21.0.3
v  ~StubRoutines::call_stub 0x000001eb61fa100d
```
It looks like you have some serious memory corruption issue going on, I'd recommend trying a different java version (maybe java 17), and if that is also broken check your computer RAM - it might be a hardware issue

**Comment by CrazyNutLz** [comment]
> ```
> J 795  jdk.internal.misc.Unsafe.allocateInstance(Ljava/lang/Class;)Ljava/lang/Object; java.base@21.0.3 (0 bytes) @ 0x000001eb627303a7 [0x000001eb62730340+0x0000000000000067]
> J 23661 c1 gregtech.api.util.GT_ModHandler.lambda$bulkRemoveByRecipe$2(Ljava/util/List;Lnet/minecraft/item/crafting/IRecipe;)Z (39 bytes) @ 0x000001eb5d521864 [0x000001eb5d5212a0+0x00000000000005c4]
> J 23662 c1 gregtech.api.util.GT_ModHandler$$Lambda+0x000001eb0355ac50.test(Ljava/lang/Object;)Z (12 bytes) @ 0x000001eb5d5225c4 [0x000001eb5d522440+0x0000000000000184]
> J 10193 c2 java.util.stream.ReferencePipeline$2$1.accept(Ljava/lang/Object;)V java.base@21.0.3 (27 bytes) @ 0x000001eb63361880 [0x000001eb63361820+0x0000000000000060]
> J 19400 c2 java.util.ArrayList$ArrayListSpliterator.forEachRemaining(Ljava/util/function/Consumer;)V java.base@21.0.3 (127 bytes) @ 0x000001eb63b3acac [0x000001eb63b3abc0+0x00000000000000ec]
> J 11073 c2 java.util.stream.AbstractPipeline.wrapAndCopyInto(Ljava/util/stream/Sink;Ljava/util/Spliterator;)Ljava/util/stream/Sink; java.base@21.0.3 (18 bytes) @ 0x000001eb62ba17d4 [0x000001eb62ba1660+0x0000000000000174]
> J 23484 c1 java.util.stream.ReduceOps$ReduceTask.doLeaf()Ljava/lang/Object; java.base@21.0.3 (5 bytes) @ 0x000001eb5c84a50c [0x000001eb5c84a2e0+0x000000000000022c]
> J 23487 c1 java.util.stream.AbstractTask.compute()V java.base@21.0.3 (141 bytes) @ 0x000001eb5ad9af64 [0x000001eb5ad9a800+0x0000000000000764]
> J 14111 c1 java.util.concurrent.CountedCompleter.exec()Z java.base@21.0.3 (6 bytes) @ 0x000001eb5bf46944 [0x000001eb5bf46840+0x0000000000000104]
> J 23480 c1 java.util.concurrent.ForkJoinTask.doExec()I java.base@21.0.3 (37 bytes) @ 0x000001eb5ba281cc [0x000001eb5ba28080+0x000000000000014c]
> j  java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Ljava/util/concurrent/ForkJoinTask;Ljava/util/concurrent/ForkJoinPool$WorkQueue;)V+19 java.base@21.0.3
> J 14112 c1 java.util.concurrent.ForkJoinPool.scan(Ljava/util/concurrent/ForkJoinPool$WorkQueue;II)I java.base@21.0.3 (263 bytes) @ 0x000001eb5c169274 [0x000001eb5c168c40+0x0000000000000634]
> j  java.util.concurrent.ForkJoinPool.runWorker(Ljava/util/concurrent/ForkJoinPool$WorkQueue;)V+35 java.base@21.0.3
> j  java.util.concurrent.ForkJoinWorkerThread.run()V+31 java.base@21.0.3
> v  ~StubRoutines::call_stub 0x000001eb61fa100d
> ```
> 
> It looks like you have some serious memory corruption issue going on, I'd recommend trying a different java version (maybe java 17), and if that is also broken check your computer RAM - it might be a hardware issue

I switched to Java 17, but the same issue persists. I ran various memory testing software on the server, and they all showed no problems. To rule out hardware issues, I tried running the server on my own computer, but it still failed to run. It didn't even start successfully once... It's very strange. Below is the log from running the server on my own computer.

[cmdlog.txt](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/files/15203552/cmdlog.txt)
[crash-2024-05-04_01.59.44-server.txt](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/files/15203555/crash-2024-05-04_01.59.44-server.txt)

**Comment by CrazyNutLz** [comment]
I seem to have solved the error, but it's very mystical... I don't know what kind of weird bug it is.

I tried for a long time on two computers, and it was still the same problem.
In desperation, I tried to start the previous 2.4.0 server again, and found that the old server also encountered the same error (which had never happened before).
I thought about it, and the only issue was that I had installed Java 21 later to run the 2.6.0 server.
So, I tried moving the newly installed Java 21 to a separate folder (no longer in the same parent directory as Java 8).
After doing this, I tested launching on both computers about 20 times simultaneously, and the error did not occur again.

Although I'm not sure if my operation was the reason for the successful start...
But at least the problem is solved. I hope if someone else encounters the same issue, this might provide a somewhat unreliable reference...
Thank you very much for your help.
